Bluewire.Tools.Runner/FindCommits/ToolRunner.cs

```diff
@@ -0,0 +1,182 @@
+
+namespace Bluewire.Tools.Runner.FindCommits
+{
+    public class ToolRunner : IToolRunner
+    {
+        public string Name => "find-commits";
+
+        public void Describe(TextWriter writer)
+        {
+            writer.WriteLine($"  {Name}: Find the commit hash(es) for the specified semantic version (release, build number and optional semantic tag).");
+        }
+
+        private static OptionSet CreateOptions(Arguments arguments)
+        {
+            return new OptionSet
+            {
+                {"s=|semver=", "Resolve a commit hash for a semantic version (<major>.<minor>.<build>[-semtag]). Omit the semtag if you want all semantic version tags to be searched.", o => arguments.Request(RequestType.SemanticVersion, o) },
+                {"repo=|repository=", "Specify the repository to use. Default: current directory.", o => arguments.WorkingCopyOrRepo = o}
+            };
+        }
+
+        public int RunMain(string[] args, string parentArgs)
+        {
+            var arguments = new Arguments();
+            var options = CreateOptions(arguments);
+            var sessionArgs = new SessionArguments<Arguments>(arguments, options);
+            sessionArgs.Application += parentArgs;
+            return new ConsoleSession<Arguments>(sessionArgs).Run(args, async a => await new Impl(a).Run());
+        }
+
+        class Impl
+        {
+            private readonly Arguments arguments;
+
+            public Impl(Arguments arguments)
+            {
+                this.arguments = arguments;
+
+                Log.Configure();
+                Log.SetConsoleVerbosity(arguments.Verbosity);
+            }
+
+            public async Task<int> Run()
+            {
+                TryInferArgumentsFromList(arguments);
+
+                var job = CreateJob(arguments);
+
+                var git = await new GitFinder().FromEnvironment();
+                var gitSession = new GitSession(git, new ConsoleInvocationLogger());
+
+                var gitRepository = GetGitRepository(arguments.WorkingCopyOrRepo);
+
+                var builds = await job.ResolveCommits(gitSession, gitRepository);
+                
+                RenderHighestPriorityResult(builds);
+
+                if (builds.Length == 0)
+                    return 1;
+                else if (builds.Length > 1)
+                    return 2;
+                return 0;
+            }
+
+            private static void RenderHighestPriorityResult(Shared.Build[] builds)
+            {
+                if (builds.Length == 0)
+                {
+                    Console.Error.WriteLine("No commits found for that build. Try running 'git fetch' to update your repo.");
+                    return;
+                }
+
+                // Don't assume that multiple matching commits are delivered to us in the correct priority order
+                var releaseBuild = builds.Where(build => build.SemanticVersion.SemanticTag == "release").SingleOrDefault();

```

Comment:
I'd be inclined to define a Comparer and use a sort. Also, I thought we were going to render all unique matches with their 'best' tag?